cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -201,11 +208,19 @@ private void executeAction(ApplicationSpecification appSpec, WorkflowActionNode
 
     status.put(node.getNodeId(), node);
 
-    WorkflowAction action = initialize(actionSpec, classLoader, instantiator, token);
+    final WorkflowAction action = initialize(actionSpec, classLoader, instantiator, token);
     try {
       ClassLoader oldClassLoader = ClassLoaders.setContextClassLoader(action.getClass().getClassLoader());
```

Comment:
Since you moved the execution into an Executor, setting of the context classloader should be moved inside the `Runnable` as well.